src/main/java/com/mixpanel/android/viewcrawler/ViewVisitor.java

```diff
@@ -119,11 +123,35 @@ protected String name() {
         private final Object[] mOriginalValueHolder;
     }
 
+    public static class LayoutUpdateException extends IllegalStateException {
+        public LayoutUpdateException(String message, JSONObject info) {
+            super(message);
+            mErrorInfo = info;
+        }
+
+        public JSONObject getErrorInfo() {
```

Comment:
Let's not use JSONObjects unless we're doing I/O or serializing things to storage. If we can, it would be great to use types.